lib/sudo-exec/src/monitor.rs

```diff
@@ -111,13 +117,13 @@ impl MonitorRelay {
             // Kill the command with increasing urgency.
             SIGALRM => {
                 // Based on `terminate_command`.
-                kill(self.command_pid, SIGHUP).ok();
-                kill(self.command_pid, SIGTERM).ok();
+                kill(self.command_pid.unwrap_or(-1), SIGHUP).ok();
```

Comment:
~~could you refactor this `-1` into a constant for readability? I presume it means "self"?~~ actually `man kill` says

> If pid equals -1, then sig is sent to every process for which the  calling process has permission to send signals, except for process 1 (init), but see below.

that sounds too far reaching. shouldn't this be constrained to the process group? `man kill` says

> If pid is less than -1, then sig is sent to every process in the process group whose ID is -pid.

Reply:
This will be solved by the refactor as `self.command_pid` doesn't need to be `Option<ProcessId>` yet and we know that it won't be negative. But yes, this section of code should check that we don't do that. Maybe our `ProcessId` should be unsigned after all.